src/variety/railpool.js

```diff
@@ -0,0 +1,659 @@
+			var result = [];
+			for (var i = 0; i < this.length; i++) {
+				var cell = this[i];
+				for (var k = 0; k < cell.qnums.length; k++) {
+					var num = cell.qnums[k];
+					if (num === -2 || !result.includes(num)) {
+						result.push(num);
+					}
+				}
+			}
+			return result;
+		},
+		getSegmentLengthsSet: function() {
+			var result = [];
+			for (var i = 0; i < this.length; i++) {
+				var cell = this[i];
+
+				var horiz = cell.getSegment(true);
+				var vert = cell.getSegment(false);
+
+				if (horiz.length > 0 && !result.includes(horiz.length)) {
+					result.push(horiz.length);
+				}
+
+				if (vert.length > 0 && !result.includes(vert.length)) {
+					result.push(vert.length);
+				}
+			}
+			return result;
+		}
+	},
+	"ObjectOperation2:Operation": {
+		setData: function(cell, old, val) {
+			this.bx = cell.bx;
+			this.by = cell.by;
+			this.old = old;
+			this.val = val;
+			this.property = "qnums";
+		},
+		decode: function(strs) {
+			if (strs.shift() !== "CR") {
+				return false;
+			}
+			this.bx = +strs.shift();
+			this.by = +strs.shift();
+			var str = strs.join(",");
+			var strs2 = str.substr(1, str.length - 2).split(/\],\[/);
+			if (strs2[0].length === 0) {
+				this.old = [];
+			} else {
+				this.old = strs2[0].split(/,/);
+				for (var i = 0; i < this.old.length; i++) {
+					this.old[i] = +this.old[i];
+				}
+			}
+			if (strs2[1].length === 0) {
+				this.val = [];
+			} else {
+				this.val = strs2[1].split(/,/);
+				for (var i = 0; i < this.val.length; i++) {
+					this.val[i] = +this.val[i];
+				}
+			}
+			return true;
+		},
+		toString: function() {
+			return [
+				"CR",
+				this.bx,
+				this.by,
+				"[" + this.old.join(",") + "]",
+				"[" + this.val.join(",") + "]"
+			].join(",");
+		},
+
+		isModify: function(lastope) {
+			// 前回と同じ場所なら前回の更新のみ
+			if (
+				lastope.property === this.property &&
+				lastope.bx === this.bx &&
+				lastope.by === this.by &&
+				this.puzzle.pzpr.util.sameArray(lastope.val, this.old)
+			) {
+				lastope.val = this.val;
+				return true;
+			}
+			return false;
+		},
+
+		undo: function() {
+			this.exec(this.old);
+		},
+		redo: function() {
+			this.exec(this.val);
+		},
+		exec: function(val) {
+			var puzzle = this.puzzle,
+				cell = puzzle.board.getc(this.bx, this.by);
+			cell.setQnums(val);
+			cell.draw();
+			puzzle.checker.resetCache();
+		}
+	},
+	OperationManager: {
+		addExtraOperation: function() {
+			this.operationlist.push(this.klass.ObjectOperation2);
+		}
+	},
+
+	Border: {
+		enableLineNG: true
+	},
+	Board: {
+		hasborder: 1
+	},
+	AreaRoomGraph: {
+		enabled: true
+	},
+	LineGraph: {
+		enabled: true
+	},
+
+	//---------------------------------------------------------
+	// 画像表示系
+	Graphic: {
+		irowake: true,
+
+		// individual number error coloring inspired on lohkous.js
+		getQuesNumberColor: function(cell, i) {
+			if (cell.error & 1) {
+				return this.errcolor1;
+			} else if (cell.error & (8 << i)) {
+				return "red";
```

Comment:
It's more common to use `this.errcolor1` for errors instead of bright red, but I'm not sure if I should change Lohkous to use `red` instead...

Reply:
I'm using `red` to emphasize that it's a problem with that individual number and not the whole cell. Right now all the other correct numbers will be `this.errcolor1` since that's the usual color for a number over an errored cell, so there are two options:
 - `this.errcolor1` for all and `red` for the specific number (as it is right now)
 - gray/black for all and  `this.errcolor1` for the specific number

Whichever you prefer
![options](https://user-images.githubusercontent.com/10251980/168684241-f15a1244-4389-4917-a171-75de17873788.png)

Reply:
Keeping the other numbers black looks good to me.